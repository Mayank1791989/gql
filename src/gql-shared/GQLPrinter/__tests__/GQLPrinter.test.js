/* @flow */
import { parse, getDescription } from 'graphql';
import { printASTNode } from '../index';

function print(text: string, commentDescriptions: boolean) {
  const astNode = parse(text);
  const [node] = astNode.definitions;
  return printASTNode(node, getDescription(node, { commentDescriptions }), {
    commentDescriptions,
  });
}

describe('old style Descriptions', () => {
  it('commentDescription: true', () => {
    expect(
      print(
        `
        # old style comment description
        type Test { id: string }
      `,
        true,
      ),
    ).toMatchSnapshot();
  });

  it('commentDescription: false', () => {
    expect(
      print(
        `
        # old style comment description
        type Test { id: string }
      `,
        false,
      ),
    ).toMatchSnapshot();
  });
});

describe('new style Descriptions', () => {
  it('commentDescription: true', () => {
    expect(
      print(
        `
        " new style comment "
        type Test { id: string }
      `,
        true,
      ),
    ).toMatchSnapshot();
  });

  it('commentDescription: false', () => {
    expect(
      print(
        `
        "new style comment"
        type Test { id: string }
      `,
        false,
      ),
    ).toMatchSnapshot();
  });
});
