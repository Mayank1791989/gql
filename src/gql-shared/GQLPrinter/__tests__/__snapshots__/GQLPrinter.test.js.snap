// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new style Descriptions commentDescription: false 1`] = `
"\\"new style comment\\"
        type Test { id: string }"
`;

exports[`new style Descriptions commentDescription: true 1`] = `
"\\" new style comment \\"
        type Test { id: string }"
`;

exports[`old style Descriptions commentDescription: false 1`] = `"type Test { id: string }"`;

exports[`old style Descriptions commentDescription: true 1`] = `
"# old style comment description
type Test { id: string }"
`;
